src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/__init__.py

```diff
@@ -5,105 +5,186 @@
-    while True:
-        ans = input(MESSAGES_GLOBAL_SETTINGS_EXISTS)
-        if not ans or ans.lower() == 'n':
-            return False
-        if ans.lower() == 'y':
-            return True
-
 def _print_cur_configuration(file_config):
+    print(MSG_HEADING_CURRENT_CONFIG_INFO)
     for section in file_config.sections():
+        print()
         print('[{}]'.format(section))
         for option in file_config.options(section):
             print('{} = {}'.format(option, file_config.get(section, option)))
-        print()
     env_vars = [ev for ev in os.environ if ev.startswith(ENV_VAR_PREFIX)]
     if env_vars:
-        print(MESSAGES_ENV_VARS_HEADING)
+        print(MSG_HEADING_ENV_VARS)
         print('\n'.join(['{} = {}'.format(ev, os.environ[ev]) for ev in env_vars]))
-        print()
+
+def _get_envs():
+    if os.path.isdir(ENV_CONFIG_DIR):
+        return [f for f in os.listdir(ENV_CONFIG_DIR) \
+                if os.path.isfile(os.path.join(ENV_CONFIG_DIR, f))]
+    return []
+
+def _config_env_public_azure(_):
+    from azure.cli.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.resource.resources import ResourceManagementClient
+    from azure.cli._profile import Profile
+    # Determine if user logged in
+    try:
+        list(get_mgmt_service_client(ResourceManagementClient).resources.list())
+    except CLIError:
+        # Not logged in
+        import getpass
+        login_successful = False
+        while not login_successful:
+            method_index = prompt_choice_list(MSG_PROMPT_LOGIN, LOGIN_METHOD_LIST)
+            profile = Profile()
+            interactive = False
+            username = None
+            password = None
+            service_principal = None
+            tenant = None
+            if method_index == 0: # device auth
+                interactive = True
+            elif method_index == 1: # username and password
+                username = input('Username: ')
+                password = getpass.getpass('Password: ')
+            elif method_index == 2: # service principal with secret
+                service_principal = True
+                username = input('Service principal: ')
+                tenant = input('Tenant: ')
+                password = getpass.getpass('Client secret: ')
+            elif method_index == 3: # skip
+                return
+            try:
+                profile.find_subscriptions_on_login(
+                    interactive,
+                    username,
+                    password,
+                    service_principal,
+                    tenant)
+                login_successful = True
+                logger.warning('Login successful!')
+            except AdalError as err:
+                logger.error('Login error!')
+                logger.error(err)
+
+def _create_or_update_env(env_name=None):
+    if not env_name:
+        # TODO Support new env creation
+        print('This feature is coming soon.\n')
+        sys.exit()
```

Comment:
I think the exit code should be 1. This is more or less looks like .net's  'not yet implementation'.
Up to you since this is temporary anyway